docs/cookbook/building-navigation.md

```diff
@@ -0,0 +1,292 @@
+  </template>
+</AposCodeBlock>
+
+By default the page data object includes all properties from the database document. We can **limit that returned data** (for a minor speed improvement and clearer logging) by adding a projection to the query builder. *Make sure to include the `path` in this projection to avoid crashing.*
+
+<AposCodeBlock>
+  ```javascript
+  module.exports = {
+    options: {
+      builders: {
+        children: true,
+        ancestors: {
+          children: {
+            depth: 2,
+            project: {
+              title: 1,
+              path: 1,
+              _url: 1,
+              _children: 1
+            }
+          }
+        }
+      }
+    }
+  }
+  ```
+  <template v-slot:caption>
+    modules/@apostrophecms/page/index.js
+  </template>
+</AposCodeBlock>
+
+If pages had **thumbnail images we wanted to show in the navigation**, we can include that as well. If we had added an area field called `thumbnail` to the page schema, we could get that data by also *configuring the query builder to include that area data*. (The relationship field data from the image widget isn't stored directly on the page.)
+
+<AposCodeBlock>
+  ```javascript
+  module.exports = {
+    options: {
+      builders: {
+        children: true,
+        ancestors: {
+          children: {
+            depth: 2,
+            project: {
+              title: 1,
+              path: 1,
+              thumbnail: 1, // 👈 Also included in the projection
+              _url: 1,
+              _children: 1
+            },
+            // 👇 Now including the area data
+            areas: [ 'thumbnail' ]
+          }
+        }
+      }
+    }
+  }
+  ```
+  <template v-slot:caption>
+    modules/@apostrophecms/page/index.js
+  </template>
+</AposCodeBlock>
+
+We would then display the image [using an `area` tag](/guide/media.html#the-image-widget-option) or [the template methods for accessing the image attachment](/guide/media.html#the-relationship-field-option).
+
+As you can see, page templates come ready with page tree data that is ready to become site navigation. With additional configuration we can customize the data that templates give us. The examples above are only a few such ways to configure it.
+
+## Add fields for manual nav building
+
+The other common approach to site navigation is to let editors build it manually. They would be able to choose the specific page links and the order in which the links should appear. Since the site nav is the same across every page (in most cases), we store the navigation data in [global settings](/guide/global.md).
+
+Before we look at a code example, let's think about how we would want to define fields for site navigation.
+
+- Website navigation is an order list of links, some of which might contain their own list of links (second-level navigation). This translates well to an array structure, so **we use an [`array` field type](/reference/field-types/array.md)**.
```

Comment:
I would say "the" rather than "an" here. I'm sure this is the kind of review you needed from me, amirite?

Reply:
yep 😃 I think it was supposed to be "an array field." I'll fix that.